CMakeLists.txt

```diff
@@ -51,7 +51,7 @@ else()
 endif()
 
 # Configure preprocessor definitions
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/config.hpp.in" "${CMAKE_CURRENT_BINARY_DIR}/include/${PROJECT_NAME}/config.hpp")
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/config.hpp.in" "${CATKIN_DEVEL_PREFIX}/include/${PROJECT_NAME}/config.hpp")
```

Comment:
Not sure what is best here (also below for the install rule). Maybe @meyerj has a better idea?

Reply:
I have not checked everything else, but the general rule is:

- Generate files in the devel-space (`CATKIN_DEVEL_PREFIX`) if they are needed by other packages that depend on `cilantro` (which seems to be the case here unless `config.hpp` is only included internally from source files in this package)
  - Do not forget to `include_directories(${CATKIN_DEVEL_PREFIX}/include)` and `catkin_package(INCLUDE_DIRS ${CATKIN_DEVEL_PREFIX}/include)`. Both are not added automatically by catkin macros.
- Generate files in the build-space (`CMAKE_CURRENT_BINARY_DIR`) if they are purely internal
  - In some cases `catkin` itself generates temporary files in the build-space, but then they additionally have to be copied to devel-space and/or installed.

Reply:
Thanks for the explanation @meyerj .

I did what you recommended above except that I missed the `catkin_package(INCLUDE_DIRS ${CATKIN_DEVEL_PREFIX}/include)`. Strangely, it did not fail to compile, any idea why?

Reply:
Typically the other package also uses a `*_msgs` package or something else from the same devel-space that already adds the include dir to the search path.

Reply:
Thanks for the explanation! Always good to learn.